akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingSingleShardPerEntitySpec.scala

```diff
@@ -0,0 +1,138 @@
+    akka.cluster.sharding.updating-state-timeout = 1s
+    """).withFallback(MultiNodeClusterSpec.clusterConfig))
+
+  testTransport(on = true)
+}
+
+class ClusterShardingSingleShardPerEntitySpecMultiJvmNode1 extends ClusterShardingSingleShardPerEntitySpec
+class ClusterShardingSingleShardPerEntitySpecMultiJvmNode2 extends ClusterShardingSingleShardPerEntitySpec
+class ClusterShardingSingleShardPerEntitySpecMultiJvmNode3 extends ClusterShardingSingleShardPerEntitySpec
+class ClusterShardingSingleShardPerEntitySpecMultiJvmNode4 extends ClusterShardingSingleShardPerEntitySpec
+class ClusterShardingSingleShardPerEntitySpecMultiJvmNode5 extends ClusterShardingSingleShardPerEntitySpec
+
+abstract class ClusterShardingSingleShardPerEntitySpec extends MultiNodeSpec(ClusterShardingSingleShardPerEntitySpecConfig)
+  with STMultiNodeSpec with ImplicitSender {
+  import ClusterShardingSingleShardPerEntitySpec._
+  import ClusterShardingSingleShardPerEntitySpecConfig._
+
+  override def initialParticipants = roles.size
+
+  def join(from: RoleName, to: RoleName): Unit = {
+    runOn(from) {
+      Cluster(system) join node(to).address
+      startSharding()
+    }
+    enterBarrier(from.name + "-joined")
+  }
+
+  def startSharding(): Unit = {
+    ClusterSharding(system).start(
+      typeName = "Entity",
+      entityProps = Props[Entity],
+      settings = ClusterShardingSettings(system),
+      extractEntityId = extractEntityId,
+      extractShardId = extractShardId)
+  }
+
+  lazy val region = ClusterSharding(system).shardRegion("Entity")
+
+  def joinAndAllocate(node: RoleName, entityId: Int): Unit = {
+    within(10.seconds) {
+      join(node, first)
+      runOn(node) {
+        region ! entityId
+        expectMsg(entityId)
+        lastSender.path should be(region.path / s"$entityId" / s"$entityId")
+      }
+    }
+    enterBarrier(s"started-$entityId")
+  }
+
+  s"Cluster sharding with single shard per entity" must {
+
+    "use specified region" in {
+      joinAndAllocate(first, 1)
+      joinAndAllocate(second, 2)
+      joinAndAllocate(third, 3)
+      joinAndAllocate(fourth, 4)
+      joinAndAllocate(fifth, 5)
+
+      runOn(first) {
+        // coordinator is on 'first', blackhole 3 other means that it can't update with WriteMajority
+        testConductor.blackhole(first, third, Direction.Both).await
+        testConductor.blackhole(first, fourth, Direction.Both).await
+        testConductor.blackhole(first, fifth, Direction.Both).await
+
+        // shard 6 not allocated yet and due to the blackhole it will not be completed
+        region ! 6
+
+        // shard 1 location is know by 'first' region, not involving coordinator
+        region ! 1
+        expectMsg(1)
+
+        // shard 2 location not known at 'first' region yet, but coordinator is on 'first' and should
+        // be able to answer GetShardHome even though previous request for shard 4 has not completed yet
+        region ! 2
+        expectMsg(2)
```

Comment:
this test would previously timeout here, because it was waiting on update for 6 to be completed